src/lib.rs

```diff
@@ -195,6 +177,40 @@ fn sha256_of_descriptor(desc: &Descriptor) -> Result<&str> {
 }
 
 impl OciDir {
+    /// Create an empty config descriptor.
+    /// See https://github.com/opencontainers/image-spec/blob/main/manifest.md#guidance-for-an-empty-descriptor
+    /// Our API right now always mutates a manifest, which means we need
+    /// a "valid" manifest, which requires a "valid" config descriptor.
+    fn empty_config_descriptor(&self) -> Result<oci_image::Descriptor> {
+        let empty_descriptor = oci_image::DescriptorBuilder::default()
+            .media_type(MediaType::EmptyJSON)
+            .size(2_u32)
+            .digest(Sha256Digest::from_str(
+                "44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
+            )?)
+            .data("e30=")
+            .build()?;
+
+        if !self
+            .dir
+            .exists(OciDir::parse_descriptor_to_path(&empty_descriptor)?)
```

Comment:
Hmm actually I think the intent of adding a dedicated media type for this is that we don't need to store it, we could just synthesize it on demand right?

I guess if we don't store it then people calling `read_json_blob` on that type would get an error...but we can just tell them "don't do that"?

Reply:
Well the reason for writing it to the disk is that I currently use an empty config for PuzzleFS, but skopeo won't copy the image if it detects that not all the descriptors are present in 'blobs/sha256' (even though I store the data inline, in the descriptor, as a base64 encoded empty json). So without writing the blob we end up with an image that's not conformant to the spec.

Reply:
Right, sorry for making you repeat that. I think that's arguably a minor bug in skopeo but yep let's be compatible with it!